bin/BuildPackages.sh

```diff
@@ -21,46 +29,48 @@ set -e
 # an idea what to do for a complete installation of GAP.
 
 
-if [ $# -eq 0 ]
-  then
-    echo "Assuming default GAP location: ../.."
-    GAPDIR=../..
-  else
-    echo "Using GAP location: $1"
-    GAPDIR="$1"
+if [[ "$#" -eq 0 ]]
+then
+  echo "Assuming default GAP location: ../.."
+  GAPDIR="../.."
+else
+  echo "Using GAP location: $1"
+  GAPDIR="$1"
 fi
 
 # Is someone trying to run us from inside the 'bin' directory?
-if [ -f gapicon.bmp ]
-  then
-    echo "This script must be run from inside the pkg directory"
-    echo "Type: cd ../pkg; ../bin/BuildPackages.sh"
-    exit 1
+if [[ -f "gapicon.bmp" ]]
+then
+  echo "This script must be run from inside the pkg directory"
+  echo "Type: cd ../pkg; ../bin/BuildPackages.sh"
+  exit 1
 fi
 
 # We need any subdirectory, to test if $GAPDIR is right
-SUBDIR=`ls -d */ | head -n 1`
-if ! (cd $SUBDIR && [ -f $GAPDIR/sysinfo.gap ])
-  then
-    echo "$GAPDIR is not the root of a gap installation (no sysinfo.gap)"
-    echo "Please provide the absolute path of your GAP root directory as"
-    echo "first argument to this script."
-    exit 1
+SUBDIR="$(find -L . -maxdepth 1 -mindepth 1 -type d | head -n 1)"
+if ! (cd "$SUBDIR" && [[ -f "$GAPDIR/sysinfo.gap" ]])
+then
+  echo "$GAPDIR is not the root of a gap installation (no sysinfo.gap)"
+  echo "Please provide the absolute path of your GAP root directory as"
+  echo "first argument to this script."
+  exit 1
 fi
 
-if (cd $SUBDIR && grep 'ABI_CFLAGS=-m32' $GAPDIR/Makefile > /dev/null) ; then
+if (cd "$SUBDIR" && grep 'ABI_CFLAGS=-m32' $GAPDIR/Makefile > /dev/null)
+then
   echo "Building with 32-bit ABI"
-  ABI32=YES
+  ABI32="YES"
   CONFIGFLAGS="CFLAGS=-m32 LDFLAGS=-m32 LOPTS=-m32 CXXFLAGS=-m32"
-fi;
+fi
 
 # Many package require GNU make. So use gmake if available,
 # for improved compatibility with *BSD systems where "make"
 # is BSD make, not GNU make.
-if ! [ x`which gmake` = "x" ] ; then
-  MAKE=gmake
+if ! [[ "x$(which gmake)" = "x" ]]
+then
+  MAKE="gmake"
 else
-  MAKE=make
+  MAKE="make"
```

Comment:
Why did you add all those superfluous quotes? I mean, they make sense around paths (which could contain spaces), so in particular for GAPDIR. But for constants where we know that quoting is no issue, this seems a bit iffy...